.github/workflows/ci.yml

```diff
@@ -0,0 +1,275 @@
+name: Gazebo CI
+
+on:
+  push:
+    tags:
+      - self-hosted-*
+    branches:
+      - main
+      - staging
+  pull_request:
+
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+env:
+  total-runners: 6
+
+jobs:
+  install:
+    name: Install deps
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '18.8.0'
+      - name: Cache node_modules
+        id: cache-node-modules
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-gazebo-node-modules
+        with:
+          path: |
+            node_modules
+          key: ${{ runner.os }}-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ env.cache-name }}-
+
+      - name: Install packages
+        if: ${{ steps.cache-node-modules.outputs.cache-hit != 'true' }}
+        run: |
+          npm ci
```

Comment:
I would recommend trying out [this](https://bun.sh/guides/install/cicd) if the cache is missed, should be a lot faster install time over `npm`, and outputs the same `node_modules` directory